### PR TITLE
Resolve common static analysis issues

### PR DESCRIPTION
**Pull request by timtebeek**
Via: https://docs.openrewrite.org/reference/recipes/java/cleanup/commonstaticanalysis

Disabled:
org.openrewrite.java.cleanup.AddSerialVersionUidToSerializable
org.openrewrite.java.cleanup.ControlFlowIndentation
org.openrewrite.java.cleanup.EqualsAvoidsNull
org.openrewrite.java.cleanup.ExplicitInitialization
org.openrewrite.java.cleanup.HiddenField
org.openrewrite.java.cleanup.HideUtilityClassConstructor
org.openrewrite.java.cleanup.MethodNameCasing
org.openrewrite.java.cleanup.NeedBraces
org.openrewrite.java.cleanup.UnnecessaryThrows

### CODE REVIEW

**Comment by timtebeek** [comment]
Some of these are perhaps a bit nitpicky, but overall they help the code readability. I had disabled a few that caused too many changes for little added value.

Also the added `final` here and there might technically be breaking a public API. I leave it up to you whether those classes were actually intended to be open for extension/override or not.

**Comment by timtebeek** [comment]
@jamietanna Would you mind reviewing this one? :)

**Review by jamietanna — COMMENTED** [review]
For the classes now being marked as `final`, does this have an impact we'll need to be careful of / at least call out in the release notes for consumers mocking/extending them?

File: src/test/java/com/github/tomakehurst/wiremock/BindAddressTest.java [hunk 1]
```diff
@@ -78,23 +79,16 @@ public void stop() {
   @Test
   public void shouldRespondInTheBindAddressOnlyOnHttp() throws Exception {
     executeGetIn(localhost);
-    try {
-      executeGetIn(nonBindAddress);
-      fail("Should not accept HTTP connection to [" + nonBindAddress + "]");
-    } catch (Exception ex) {
-    }
+
+    assertThrows(RuntimeException.class, () -> executeGetIn(nonBindAddress));
```
Comment (jamietanna) [inline src/test/java/com/github/tomakehurst/wiremock/BindAddressTest.java:83]: We've changed the intent of this test, right? Should be:

```suggestion
    assertThrows(Exception.class, () -> executeGetIn(nonBindAddress));
```

To match the previous intent?

File: src/test/java/com/github/tomakehurst/wiremock/LogTimingAcceptanceTest.java [hunk 2]
```diff
@@ -54,12 +54,12 @@ public void serveEventIncludesTotalAndServeDuration() {
 
   @Test
   public void includesAddedDelayInTotalWhenAsync() {
-    final int DELAY = 500;
+    final int delay = 500;
```
Comment (jamietanna) [inline src/test/java/com/github/tomakehurst/wiremock/LogTimingAcceptanceTest.java:57]: We should probably follow up and remove the `final`s from local varialbes, as I don't think we want them

File: src/test/java/com/github/tomakehurst/wiremock/matching/MockRequest.java [hunk 3]
```diff
@@ -246,20 +246,9 @@ public boolean isMultipart() {
...
-                      return true;
-                    }
-                    return false;
-                  }
-                })
-            .get()
-        : null;
+    return getParts() == null || name == null
+        ? null
+        : from(multiparts).firstMatch(input -> name.equals(input.getName())).get();
```
Comment (jamietanna) [inline src/test/java/com/github/tomakehurst/wiremock/matching/MockRequest.java:251]: I'd personally prefer we get rid of the ternary expression and replace it with an `if ... else`

**Review by jamietanna — COMMENTED** [review]
Couple of minor comments, but looking good, thanks for doing this :raised_hands:

File: src/test/java/com/github/tomakehurst/wiremock/LogTimingAcceptanceTest.java (hunk 2 above)
Comment (timtebeek) [reply src/test/java/com/github/tomakehurst/wiremock/LogTimingAcceptanceTest.java]: Done; made the diff quite a bit larger, with more potential for conflicts, but should be OK if merged soon-ish.

File: src/test/java/com/github/tomakehurst/wiremock/matching/MockRequest.java (hunk 3 above)
Comment (timtebeek) [reply src/test/java/com/github/tomakehurst/wiremock/matching/MockRequest.java]: Better indeed; changed.

**Comment by timtebeek** [comment]
Thanks for the suggestions; all applied.

As for:
> For the classes now being marked as final, does this have an impact we'll need to be careful of / at least call out in the release notes for consumers mocking/extending them?

All of these now final classes already have a private constructor:
```
src/main/java/com/github/tomakehurst/wiremock/common/DateTimeParser.java
src/main/java/com/github/tomakehurst/wiremock/common/StreamSources.java
src/main/java/com/github/tomakehurst/wiremock/common/xml/Xml.java
src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/HttpTemplateCacheKey.java
src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/RequestLine.java
src/main/java/com/github/tomakehurst/wiremock/jetty9/JettyUtils.java
com.github.tomakehurst.wiremock.common.JettySettings.Builder
com.github.tomakehurst.wiremock.extension.responsetemplating.RenderCache.Key
```

So it would only affect users that mock those classes, in which case I really wonder why.
I'd say make them final for now, and open them up again if there's a convincing case.

**Comment by jamietanna** [comment]
Sounds like a plan, thanks Tim - I didn't look that far :sweat_smile:

File: src/main/java/com/github/tomakehurst/wiremock/http/CaseInsensitiveKey.java [hunk 4]
```diff
@@ -39,9 +39,7 @@ public boolean equals(Object o) {
 
     CaseInsensitiveKey that = (CaseInsensitiveKey) o;
 
-    if (key != null ? !key.equalsIgnoreCase(that.key) : that.key != null) return false;
-
-    return true;
+    return key == null ? that.key == null : key.equalsIgnoreCase(that.key);
```
Comment (jamietanna) [inline src/main/java/com/github/tomakehurst/wiremock/http/CaseInsensitiveKey.java:42]: personally, I'd prefer this to be:

```suggestion
    if (key == null) {
      return that.key == null;
    } else {
      key.equalsIgnoreCase(that.key);
    }
```

**Comment by jamietanna** [comment]
Thanks Tim, overall changes are looking good, but I'd probably like Tom just to double check re the `final`s :+1:

File: src/main/java/com/github/tomakehurst/wiremock/http/CaseInsensitiveKey.java (hunk 4 above)
Comment (timtebeek) [reply src/main/java/com/github/tomakehurst/wiremock/http/CaseInsensitiveKey.java:42]:
```suggestion
    if (key == null) {
      return that.key == null;
    }
    return key.equalsIgnoreCase(that.key);
```
You were missing a return statement; would you be OK with this instead? I don't like unnecessarily nested else statements when we can avoid them with a return.

Comment (jamietanna) [reply src/main/java/com/github/tomakehurst/wiremock/http/CaseInsensitiveKey.java:42]: Woops, good spot.

Yep, I like that too, that'd be appreciated 👍🏽

**Comment by timtebeek** [comment]
Closing this one due to inactivity & conflicts; might want to cherry pick some of the manual fixes, and then optionally recreate the changes with the [updated common static analysis automation](https://docs.openrewrite.org/reference/recipes/java/cleanup/commonstaticanalysis).

**Pull request closed without merging**